staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1864,3 +1986,523 @@ func TestForgetWatcher(t *testing.T) {
+			opts:                      listOptions(false, nil, ""),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                      "RV=unset, allowWatchBookmarks=false, sendInitialEvents=true, expectedBookmarkRV=0",
+			opts:                      listOptions(false, pointer.Bool(true), ""),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                      "RV=unset, allowWatchBookmarks=false, sendInitialEvents=false, expectedBookmarkRV=0",
+			opts:                      listOptions(false, pointer.Bool(false), ""),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		// +-----------------+---------------------+-----------------------+
+		// | ResourceVersion | AllowWatchBookmarks | SendInitialEvents     |
+		// +=================+=====================+=======================+
+		// | 0               | true/false          | nil/true/false        |
+		// +-----------------+---------------------+-----------------------+
+		{
+			name:                      "RV=0, allowWatchBookmarks=true, sendInitialEvents=nil, expectedBookmarkRV=0",
+			opts:                      listOptions(true, nil, "0"),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                            "RV=0, allowWatchBookmarks=true, sendInitialEvents=true, expectedBookmarkRV=90",
+			opts:                            listOptions(true, pointer.Bool(true), "0"),
+			requiredResourceVersion:         100,
+			watchCacheResourceVersion:       90,
+			expectedBookmarkResourceVersion: 90,
+		},
+		{
+			name:                      "RV=0, allowWatchBookmarks=true, sendInitialEvents=false, expectedBookmarkRV=0",
+			opts:                      listOptions(true, pointer.Bool(false), "0"),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                      "RV=0, allowWatchBookmarks=false, sendInitialEvents=nil, expectedBookmarkRV=0",
+			opts:                      listOptions(false, nil, "0"),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                      "RV=0, allowWatchBookmarks=false, sendInitialEvents=true, expectedBookmarkRV=0",
+			opts:                      listOptions(false, pointer.Bool(true), "0"),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		{
+			name:                      "RV=0, allowWatchBookmarks=false, sendInitialEvents=false, expectedBookmarkRV=0",
+			opts:                      listOptions(false, pointer.Bool(false), "0"),
+			requiredResourceVersion:   100,
+			watchCacheResourceVersion: 90,
+		},
+		// +-----------------+---------------------+-----------------------+
+		// | ResourceVersion | AllowWatchBookmarks | SendInitialEvents     |
+		// +=================+=====================+=======================+
+		// | 105             | true/false          | nil/true/false        |
```

Comment:
Same comment as before - it's common to get RV that is lagging behind storage not that is in front of it.